### PR TITLE
[MSHARED-1146] Deprecate multivariant constructors for removal

### PR DESCRIPTION
**Pull request by adekzs**
The redundant constructors have been deprecated. The update is ready for review

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
Please look at build result, the same should be when you build project locally 😄

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java [hunk 1]
```diff
@@ -115,61 +115,157 @@
...
+    *
+    * @deprecated to be removed
+    * use {@link #Verifier(String basedir)}
+     *
+    * */
+    @Deprecated
     public Verifier( String basedir, boolean debug )
         throws VerificationException
     {
         this( basedir, null, debug );
```
Comment (slawekjaranowski) [inline src/main/java/org/apache/maven/shared/verifier/Verifier.java:149]: I would to change body of constructors to show replacements, so should be:
```
  this(basedir);
 ... 
 needed set methods
```
in all deprecated constructor

Comment (adekzs) [reply src/main/java/org/apache/maven/shared/verifier/Verifier.java:149]: Okay, I get your point. I will update accordingly.

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java [hunk 2]
```diff
@@ -115,87 +115,195 @@
...
+     * */
+    @Deprecated
     public Verifier( String basedir, String settingsFile, boolean debug, String[] defaultCliArguments )
         throws VerificationException
     {
-        this( basedir, settingsFile, debug, null, defaultCliArguments );
+        this( basedir );
+        setSettingsFile( settingsFile );
+        setDebug( debug );
+        setMavenHome( null );
```
Comment (slawekjaranowski) [inline src/main/java/org/apache/maven/shared/verifier/Verifier.java:201]: Is it needed with `null` ?

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java [hunk 3]
```diff
@@ -115,87 +115,195 @@
...
     public Verifier( String basedir, String settingsFile, boolean debug, boolean forkJvm, String[] defaultCliArguments )
         throws VerificationException
     {
-        this( basedir, settingsFile, debug, forkJvm, defaultCliArguments, null );
+        this( basedir );
+        setSettingsFile( basedir );
+        setDebug( debug );
+        setForkJvm( forkJvm );
+        setDefaultCliArguments( defaultCliArguments );
+        setMavenHome( null );
```
Comment (slawekjaranowski) [inline src/main/java/org/apache/maven/shared/verifier/Verifier.java:241]: the same question...

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java [hunk 4]
```diff
@@ -115,87 +115,195 @@
...
+     * and {@link #setMavenHome(String mavenHome)}  to set maven home
+     *
+     * */
+    @Deprecated
     public Verifier( String basedir, String settingsFile, boolean debug, String mavenHome )
             throws VerificationException
     {
-        this( basedir, settingsFile, debug, null, DEFAULT_CLI_ARGUMENTS, mavenHome );
+        this( basedir );
+        setForkJvm( null );
```
Comment (slawekjaranowski) [inline src/main/java/org/apache/maven/shared/verifier/Verifier.java:257]: Is it needed?

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java [hunk 5]
```diff
@@ -115,87 +115,195 @@
...
+    @Deprecated
     public Verifier( String basedir, String settingsFile, boolean debug, String mavenHome,
                      String[] defaultCliArguments )
         throws VerificationException
     {
-        this( basedir, settingsFile, debug, null, defaultCliArguments, mavenHome );
+        this( basedir );
+        setSettingsFile( settingsFile );
+        setDebug( debug );
+        setForkJvm( null );
```
Comment (slawekjaranowski) [inline src/main/java/org/apache/maven/shared/verifier/Verifier.java:280]: ?

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java [hunk 6]
```diff
@@ -115,87 +115,195 @@
...
+     * use {@link #setSettingsFile(String settingsFile)} to set settings file
+     * use {@link #setForkJvm(Boolean)}  to set forkJvm status
+     * use {@link #setDefaultCliArguments(String[] defaultCliArguments)} to set CLI Arguments
+     *  use {@link #setMavenHome(String mavenHome)}  to set maven home
+     *
+     * */
+    @Deprecated
     private Verifier( String basedir, String settingsFile, boolean debug, Boolean forkJvm, String[] defaultCliArguments,
             String mavenHome ) throws VerificationException
     {
```
Comment (slawekjaranowski) [inline src/main/java/org/apache/maven/shared/verifier/Verifier.java:299]: Is it still used?
If no we can remove private constructor at all.

Comment (adekzs) [reply src/main/java/org/apache/maven/shared/verifier/Verifier.java:299]: It is no longer in use, I will remove this.

File: src/main/java/org/apache/maven/shared/verifier/Verifier.java (hunk 5 above)
Comment (adekzs) [reply src/main/java/org/apache/maven/shared/verifier/Verifier.java:280]: setting null for ForkJvm is also unnecessary, I will remove this as well.

**Comment by slawekjaranowski** [comment]
@adekzs thanks

**Comment by jira-importer** [comment]
Resolve #143

**Comment by jira-importer** [comment]
Resolve #143

**Pull request merged**